File: app/src/main/java/omar/mohamed/socialphotoneighbour/views/SquaredImageView.java
package omar.mohamed.socialphotoneighbour.views;

import android.content.Context;
import androidx.appcompat.widget.AppCompatImageView;

public class SquaredImageView  extends AppCompatImageView {


  public SquaredImageView(Context context) {
    super(context);
  }

  @Override
  protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
    super.onMeasure(widthMeasureSpec, heightMeasureSpec);

    int uniqueValue = getMeasuredWidth();
    setMeasuredDimension(uniqueValue, uniqueValue);
  }


}

File: app/src/main/kotlin/omar/mohamed/socialphotoneighbour/views/SquaredImageView.kt
package omar.mohamed.socialphotoneighbour.views

import android.content.Context
import androidx.appcompat.widget.AppCompatImageView

class SquaredImageView(context: Context) : AppCompatImageView(context) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)

        val uniqueValue = measuredWidth
        setMeasuredDimension(uniqueValue, uniqueValue)
    }


}